Request/ParamConverter/DoctrineParamConverter.php

```diff
@@ -229,12 +267,33 @@ public function supports(ParamConverter $configuration)
 
     private function getOptions(ParamConverter $configuration)
     {
-        return array_replace(array(
+        $defaultValues = array(
             'entity_manager' => null,
             'exclude' => array(),
             'mapping' => array(),
             'strip_null' => false,
-        ), $configuration->getOptions());
+            'expr' => null,
+            'id' => null,
+            'repository_method' => null,
+            'map_method_signature' => false,
+        );
+
+        $passedOptions = $configuration->getOptions();
+
+        if (isset($passedOptions['repository_method'])) {
+            @trigger_error('The repository_method option of @ParamConverter is deprecated and will be removed in 5.0. Use the expr option or @Entity.', E_USER_DEPRECATED);
+        }
+
+        if (isset($passedOptions['map_method_signature'])) {
+            @trigger_error('The map_method_signature option of @ParamConverter is deprecated and will be removed in 5.0. Use the expr option or @Entity.', E_USER_DEPRECATED);
+        }
+
+        $extraKeys = array_diff(array_keys($passedOptions), array_keys($defaultValues));
+        if ($extraKeys) {
+            throw new \InvalidArgumentException(sprintf('Invalid option(s) passed to @ParamConverter: %s', implode(', ', $extraKeys)));
+        }
```

Comment:
I added strict validation on the `options`, which we can do safely now that this class is not extendable (everything is private).